### ISSUE TITLE
Compare with RazorLight.

### ISSUE THREAD

**Issue by MNF** [adoconnection/RazorEngineCore#58]
Could you please add to Readme comparison, why someone should choose your library compare to https://github.com/toddams/RazorLight.

Just as curiosity, what was the reason to create an alternative library, when one similar already existed?

**Comment by adoconnection** [comment]
Hi! First of all I appreciate toddams's work and it's great to have a choice of what library to use.

RazorLight as I see it, although it can render strings, it is aiming to be view engine framework detached from MVC relying on files a lot. 
Wheres my implementration aiming to be "I just want to render template from string" library. No asking any extras.

At a time I started RazorEngineCore I was looking for a library to be able:
* save / load templates -  RazorLight does not support it
* transparent caching - RazorLight have ICachingProvider but you are unable to implement anything but memory cache
* transparent compiled template handling -  once I have a template handle I  can render it at any time unlike RazorLight where I have to construct entire engine every time.
* custom template properties and methods (https://github.com/adoconnection/RazorEngineCore#helpers-and-custom-members)
* personally ```CompileRenderString``` method is making me nuts :) as a questionable decision to blindly migrate it from https://github.com/Antaris/RazorEngine 

noone was able to provide that, so I did RazorEngineCore 
Basically we do the same, but I strongly belive razor engine should work in a way RazorEngineCore do and not like RazorLight :)

**Comment by MNF** [comment]
Great answer. 
Please consider to copy to Readme or Wiki (or link from them to this answer)

**Comment by 304NotModified** [comment]
- razorlight is by default safe for producing html, as it does html encoding by default (opt out)

**Comment by benmccallum** [comment]
Is that true @adoconnection regarding no html encoding by default? That's a bit of a deal breaker for me as we've got client input being sent around by emails.

RazorLight looks a little unmaintained for my liking though, has been on rc forever. Rock and a hard place at the moment.

**Comment by 304NotModified** [comment]
> RazorLight looks a little unmaintained for my liking though, has been on rc forever. 

Also noticed that indeed.

**Comment by adoconnection** [comment]
I have updated #65 to keep HTML safety discussion in one issue.

**Comment by adoconnection** [comment]
I have started wiki page to close this issue.
all improvements are welcome!

https://github.com/adoconnection/RazorEngineCore/wiki/Package-comparison
